nexoan/crud-api/db/queryExecutor/Interfaces/executor.go

```diff
@@ -0,0 +1,454 @@
+//			"entityId": "123",
+//		},
+//	})
+type GraphExecutor interface {
+	Executor
+
+	// GraphEntity retrieves an entity's graph information.
+	//
+	// Parameters:
+	//   - ctx: Context for the operation
+	//   - plan: QueryParameter containing entity identifier
+	//
+	// Returns:
+	//   - *pb.Kind: Entity kind information
+	//   - *pb.TimeBasedValue: Entity name with temporal information
+	//   - string: Creation timestamp
+	//   - string: Termination timestamp
+	//   - error: Any error that occurred during retrieval
+	//
+	// Example:
+	//   kind, name, created, terminated, err := executor.GraphEntity(ctx, &QueryParameter{
+	//       Filters: map[string]interface{}{
+	//           "entityId": "123",
+	//       },
+	//   })
+	GraphEntity(ctx context.Context, plan *QueryParameter) (*pb.Kind, *pb.TimeBasedValue, string, string, error)
+
+	// GraphEntityCreate creates a new entity in the graph.
+	//
+	// Parameters:
+	//   - ctx: Context for the operation
+	//   - plan: QueryParameter containing entity details
+	//
+	// Returns:
+	//   - bool: True if creation was successful
+	//   - error: Any error that occurred during creation
+	//
+	// Example:
+	//   success, err := executor.GraphEntityCreate(ctx, &QueryParameter{
+	//       Operation: "create_entity",
+	//       Filters: map[string]interface{}{
+	//           "entityId": "123",
+	//           "kind": map[string]string{
+	//               "major": "Person",
+	//               "minor": "Employee",
+	//           },
+	//           "name": "John Doe",
+	//       },
+	//   })
+	GraphEntityCreate(ctx context.Context, plan *QueryParameter) (bool, error)
+
+	// GraphEntityUpdate updates an existing entity in the graph.
+	//
+	// Parameters:
+	//   - ctx: Context for the operation
+	//   - plan: QueryParameter containing update details
+	//
+	// Returns:
+	//   - bool: True if update was successful
+	//   - error: Any error that occurred during update
+	//
+	// Example:
+	//   success, err := executor.GraphEntityUpdate(ctx, &QueryParameter{
+	//       Operation: "update_entity",
+	//       Filters: map[string]interface{}{
+	//           "entityId": "123",
+	//           "name": "Jane Doe",
+	//       },
+	//   })
+	GraphEntityUpdate(ctx context.Context, plan *QueryParameter) (bool, error)
+
+	// GraphEntityFilter filters entities based on criteria.
+	//
+	// Parameters:
+	//   - ctx: Context for the operation
+	//   - plan: QueryParameter containing filter criteria
+	//
+	// Returns:
+	//   - []map[string]interface{}: List of matching entities
+	//   - error: Any error that occurred during filtering
+	//
+	// Example:
+	//   entities, err := executor.GraphEntityFilter(ctx, &QueryParameter{
+	//       Filters: map[string]interface{}{
+	//           "kind": "Person",
+	//           "department": "Engineering",
+	//       },
+	//   })
+	GraphEntityFilter(ctx context.Context, plan *QueryParameter) ([]map[string]interface{}, error)
+
+	// GraphRelationships retrieves all relationships for an entity.
```

Comment:
```suggestion
	// GraphAllRelationshipsByNode retrieves all relationships for an entity.
```

The idea is the method name should describe what it does.